dev/Dockerfile

```diff
@@ -57,37 +56,30 @@ RUN set -eux; \
     python3 \
     libpython3-dev \
     python3-pip \
+    python3.11-venv \
     python3-sphinx \
+    python3-sphinx-rtd-theme \
```

Comment:
Do we still need this explicitly (sphinx and sphinx-rtd-theme for docs), see https://github.com/apache/couchdb/blob/main/src/docs/requirements.txt?

Reply:
Good point, we bring both the mango tests and doc build dependencies in via a venv + requirements.txt file so those may not be needed. Try to to see if `make check` passes with just `python3-setuptools` and `python3.11-venv`

Reply:
Removed the extra Python packages.